# Financial Accounting 

## Chapter 1 - The Purpose and Use of Financial Statements

import the necessary libraries now then it is accessible now and for later.

In [1]:
import openpyxl  # allows to make, read in and save Excel sheets

In [2]:
import pandas as pd  # used for Dataframe operations
import numpy as np   # used for various mathematical operations

**Accounting** - the information system that identifies and records the economic events of an organization and then communicates them to a wide variety of interested users.

**accounts receivable** - the operation of receiving money in the future from customers. This is an asset as they represent an economic resource (cash) and will be eventually collected.

**accounts payable** - short term liability from some expenses when a company purchases inventory or supplies on credit from suppliers.

**interest payable** - the outstanding (unpaid) liability amounts owed to various lenders and creditors

- dividends payable, salaries payable, property payable, sales tax payable

Business goal is to sell a good or service for a price greater than the cost of producing or purchasing the good or providing the service, plus the cost of operating the business.

**net income** - revenues are greater than the expenses incurred to generate the revenues. (net earnings, profit)


------
Let us make a Python function to handle the calculation for net income (loss) so 
that it can be used repeatedly.

Python functions always start with `def` keyword, and can store various information.

In [16]:
# a function to calculate net income

def net_income(revenue: float, expenses: float): # (arguments for function):
    # formula
    net = revenue - expenses
    # print out the result with formatted string
    print(f"Net income = ${net}")

In [17]:
# call the function so we can use it
net_income(revenue= 16777, expenses= 1097)

Net income = $15680


In [18]:
net_income(1233, 1677)

Net income = $-444


### Financial Statements

There are 4 required financial statements:

1. **income statement** - reports revenues and expenses, showing how a company's operations performed during a period of time

2. **statement of changes in equity** - shows changes in each component of shareholders equity (common shares and retained earnings) and total equity during period of time

3. **statement of financial position** - what a company owns (assets), what it owes (liabilities) and resulting difference (shareholder equity) at a specific point in time

4. **statement of cash flows** - where a company obtained cash during a period of time and how that cash was used

#### Income statement

Example: Corporation Sierra has been in operation for 1 month, month ended on October 31, 2018.

Income statements typically look like this

```
Revenue
    service revenue           $20,000
Expenses
    salaries                  $5,200
    supplies                  $1,500
    rent                      $900
    depreciation              $83
    insurance                 $50
    interest                  $25            7,758
-
Income before income tax                     12,842
Income tax                                   1,800
Net income                                   $11,042
```

We can make this dataframe ourselves and deduct all the expenses together

In [20]:
sierra = pd.DataFrame (
    {
        "Revenue": [20_000] * 8, # the number of values per row must match, 7 items per row
        "Expenses": ['Service revenue','salaries','supplies','rent','depreciation','insurance','interest','income tax expense'],
        'amount': [20_000,5200,1500,900,83,50,25,1800]
    }
)

sierra

Revenue            Expenses  amount
0    20000     Service revenue   20000
1    20000            salaries    5200
2    20000            supplies    1500
3    20000                rent     900
4    20000        depreciation      83
5    20000           insurance      50
6    20000            interest      25
7    20000  income tax expense    1800

get all the expenses while not including the $20,000 revenue and create a new column that sums all of the expenses

In [25]:
# create a variable named service revenue
# dataframe, integer location, 1st row (0), 2nd column
service_rev = sierra.iloc[0][2]
service_rev

20000

In [27]:
# create a variable to store all the expenses and sum them
# from row 1 to the 7th, third column
expenses =sum( sierra.iloc[1:8,2]) 

# create a new column, subtract the revenue and store net income 
sierra['net income'] = service_rev  - expenses

sierra

Revenue            Expenses  amount  net income
0    20000     Service revenue   20000       10442
1    20000            salaries    5200       10442
2    20000            supplies    1500       10442
3    20000                rent     900       10442
4    20000        depreciation      83       10442
5    20000           insurance      50       10442
6    20000            interest      25       10442
7    20000  income tax expense    1800       10442

The net income from the example above shows 11,042 instead of $10,442

#### Statement of changes in Equity

```
common shares (end of period) = 
            common shares (begin of period) 
            + common shares issued (sold) 
            - common shares repurchased
```

In [28]:
def shareholder_equity(shares1, shares_sold, shares_repurchased):
    common_shares = (shares1 + shares_sold) - shares_repurchsed
    print(f"Common shares at end of period = ${common_shares}")